phpseclib/Common/Functions/Objects.php

```diff
@@ -0,0 +1,75 @@
+<?php
+
+/**
+ * Common Object Functions
+ *
+ * PHP version 5
+ *
+ * @category  Common
+ * @package   Functions\Objects
+ * @author    Jim Wigginton <terrafrost@php.net>
+ * @copyright 2016 Jim Wigginton
+ * @license   http://www.opensource.org/licenses/mit-license.html  MIT License
+ * @link      http://phpseclib.sourceforge.net
+ */
+
+namespace phpseclib\Common\Functions;
+
+/**
+ * Common Object Functions
+ *
+ * @package Functions\Objects
+ * @author  Jim Wigginton <terrafrost@php.net>
+ */
+abstract class Objects
```

Comment:
Okay. Thanks for the detailed explanation. I am fine with this if it is only a temporary solution. I agree that this is better than having unit tests fail permanently. An alternative would simply having properties and methods public, as you mentioned.

> So what would you recommend in that scenario?

In the long run, I would try to keep things simple only using the tools provided by PHP. That is, if `get_channel_packet` needs to be called on an instance of SSH2, `get_channel_packet` needs to be defined public.
The fact that SCP and SSH Agent need to call get_channel_packet indicates that the architecture might be suboptimal. The same problem would probably also apply to SFTP, but that is worked around by having SFTP extend SSH2. More concretely, maybe there should be a SSHChannel class that can be used by SSH2, SCP, Agent, SFTP, etc. and provides access to the necessary channel mechanisms.